bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ButtonImpl.java

```diff
@@ -46,43 +51,57 @@
 
     public static final String RESOURCE_TYPE = "core/wcm/components/button/v1/button";
 
-    private String linkURL;
-
     @Self
     private SlingHttpServletRequest request;
 
+    @ScriptVariable
+    private Resource resource;
+    
     @ScriptVariable
     private PageManager pageManager;
 
     @ValueMapValue(optional = true)
     @Named(JcrConstants.JCR_TITLE)
     private String text;
 
-    @ValueMapValue(optional = true)
-    private String link;
-
     @ValueMapValue(optional = true)
     private String icon;
 
     @ValueMapValue(optional = true)
     protected String accessibilityLabel;
 
+    @Self
+    private LinkHandler linkHandler;
+    private Link link;
```

Comment:
What about a dedicated injector for LinkHandler?

Reply:
we might think about this later, but it's not as easy as always doing the same for the same model:
* some models have different properties to read the link url from (for historical reasons)
* some models do have additional links which need to be processed with the link handler (e.g. teaser action links, image map area links, links in navigation-like models etc.)

so i currently would prefer to keep it like this until reocurring patterns emerge. after all, it's still an internal API and not yet published, so we can change it at any time.

Reply:
I was thinking about an annotation with fields which allow to override the node and the properties from which to read, but I agree, we can do this later on...